### PR TITLE
Access Blueocean console from testplan view in webapp

### PR DESCRIPTION
**Pull request by sameerawickramasekara**
**Purpose**

provide a blue ocean link to the user in the test plan view

**Approach**
link is provided as an input parameter to the RunTestPlan command and then it is saved in the database.

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### IMPORTANT!!

- from this PR the create-or-extend-tables property is added to the persistant xml. JPA will modify the existing table in runtime and add new columns. so there is **no need of a manual database migration script.**

- Since this PR introduces Database level changes any jobs running during the PR merge will not revcieve these changes and will break during execution. Hence its **important that no builds are running until the deployments are updated.**

### CODE REVIEW

File: core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java [hunk 1]
```diff
@@ -74,6 +74,11 @@
             required = true)
     private String workspace = "";
 
+    @Option(name = "--url",
+            usage = "Jenkins URL",
+            aliases = {"-u"})
```
Comment (kasunbg) [inline core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java]: Make this optional?

File: web/src/main/react-dashboard/src/components/TestRunView.js [hunk 2]
```diff
@@ -362,6 +364,16 @@ class TestRunView extends Component {
                   )}
           ><i className="fa fa-download" aria-hidden="true"> </i> Download Test-Run log</Button>
 
+          <Button id ="tdd" size="sm" style={{marginLeft: "10px"}}
```
Comment (kasunbg) [inline web/src/main/react-dashboard/src/components/TestRunView.js]: Can't we replace existing test-run.log now? This blueocean log file makes the other one obsolete isnt it?

File: core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java (hunk 1 above)
Comment (sameerawickramasekara) [reply core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java]: the default value for the required parameter is false, so this is optional. are you suggesting to make this mandatory?

File: web/src/main/react-dashboard/src/components/TestRunView.js (hunk 2 above)
Comment (sameerawickramasekara) [reply web/src/main/react-dashboard/src/components/TestRunView.js]: This change will store the blueocean urls of all the builds in the future, but for older build IMO we need to keep the existing option for a while. WDYT?

File: core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java (hunk 1 above)
Comment (kasunbg) [reply core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java]: Just wanted to make sure this is optional. (y)

File: web/src/main/react-dashboard/src/components/TestRunView.js (hunk 2 above)
Comment (kasunbg) [reply web/src/main/react-dashboard/src/components/TestRunView.js]: +1. It is confusing to have two links for almost the same thing. How can we fix that?

**Pull request merged**